docs/tutorial/04-control-snake-movements.md

```diff
@@ -0,0 +1,135 @@
+### 4. Control snake movements
+
+Awesome! Our snake moves fast tile by tile, but we cannot control it yet. Kinda boring, isn't it?
+
+In order to control the snake, we need to handle keyboard inputs when pressing any arrow key and change the snake direction accordingly.
+
+☝️ Remember, the direction is saved in the scene's state as a tuple `{x, y}`.
+
+We need to map all the arrow keys inputs into our direction system:
+
+- when pressing the left arrow key ⬅️, the snake direction needs to be set to `{-1, 0}`.
+- when pressing the right arrow key ➡️, the snake direction needs to be set to `{1, 0}`.
+- when pressing the up arrow key ⬆️, the snake direction needs to be set to `{0, -1}`.
+- when pressing the down arrow key ⬇️, the snake direction needs to be set to `{0, 1}`.
+
+And eventually, we need to ignore any other keypress input (e.g. space bar, return, etc.).
+
+Scenic conveniently provides us the [`handle_input/3`](https://hexdocs.pm/scenic/Scenic.Scene.html#c:handle_input/3) callback to implement it.
+
+> Coach: Explain what a callback is and how it is used for defining behaviours.
+
+`handle_input/3` is invoked when the scene receives an input from a driver. Then, the handler can consume the received input and either propagate it to the scene above in the ViewPort's structure or not.
+
+In our case, we will use the handler to update the direction in the scene's state without propagating the received input further since it's not needed.
+
+> Coach: Explain event bubbling and ViewPort's supervision structure.
+
+> Coach: Explain what is an input and how it [differs from events](https://hexdocs.pm/scenic/Scenic.Scene.html#module-input-vs-events).
+
+`handle_input/3` requires three arguments:
+
+- the input
```

Comment:
```suggestion
- the input message
```

Reply:
@klappradla sometimes we are using `input` and sometimes `input message`. Do you think this can lead to confusion?

Reply:
Hm, you're right. I guess we should then just simply go for `input`. The data structure behind, the "how" an input is received by the function is probably not necessary to know for people going through the tutorial.